### DISCUSSION TITLE
GNN models that output edges

### DISCUSSION THREAD

**Discussion by Fanchengyan** [pyg-team/pytorch_geometric#8146] [Q&A]
Hi all. My research content focuses on edge information. Are there any GNN models that output edges? Or can a GNN network that outputs edges be built using the MessagePassing class? I saw in the section "Passing messages between parts of the graph" in the article <https://distill.pub/2021/gnn-intro/> : that information can be passed between edges and nodes. Can we use this `MessagePassing` class to achieve the message passing jusk like in the article? I tried to read the documentation for MessagePassing but didn't understand how to operate it.

**Comment by rusty1s** [comment]
In general, GNNs are designed to compute node representations. There exists multiple ways to compute edge representations from it:
1. You use the source and destination node representations to compute the final edge representation, e.g., `[x_src, x_dst]`.
2. You operate on the line graph of your initial graph, converting the edges in your graph to nodes.
3. You convert your graph into a heterogeneous variant, in which edges are now represented as an additional node type.
